control_server.go

```diff
@@ -210,7 +219,13 @@ func (this *ControlServer) HandleVerify(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	http.Redirect(w, r, "/", http.StatusSeeOther)
+	returnSuccess(w, r)
+}
+
+func (this *ControlServer) HandleProgress(w http.ResponseWriter, r *http.Request) {
+	status := this.fetchStatus()
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(status)
```

Comment:
Is there no error for this?

Reply:
It can return an error, at this point it will have to be logging I think